libs/sr-ui/src/RelicEditor/RelicEditor.tsx

```diff
@@ -0,0 +1,328 @@
+                        .map((i) => (
+                          <Button
+                            key={i}
+                            onClick={() => update({ level: i })}
+                            disabled={!sheet || level === i}
+                          >
+                            {i}
+                          </Button>
+                        ))
+                    : null}
+                  <Button
+                    onClick={() => update({ level: level + 1 })}
+                    disabled={!sheet || level === rarity * 3}
+                  >
+                    +
+                  </Button>
+                </ButtonGroup>
+              </Box>
+
+              {/* slot */}
+              <Box component="div" display="flex">
+                <Select
+                  label="Slot"
+                  value={slotKey}
+                  onChange={(e) =>
+                    update({ slotKey: e.target.value as RelicSlotKey })
+                  }
+                  disabled={!sheet}
+                >
+                  {allRelicSlotKeys.map((s) => (
+                    <MenuItem key={s} value={s}>
+                      {s}
+                    </MenuItem>
+                  ))}
+                </Select>
+                <CardThemed bgt="light" sx={{ p: 1, ml: 1, flexGrow: 1 }}>
+                  <Suspense fallback={<Skeleton width="60%" />}>
+                    <Typography color="text.secondary">{slotKey}</Typography>
+                  </Suspense>
+                </CardThemed>
+              </Box>
+
+              {/* main stat */}
+              <Box component="div" display="flex" gap={1}>
+                <DropdownButton
+                  title={<b>{relic?.mainStatKey ?? 'Main Stat'}</b>}
+                  disabled={!sheet}
+                  color={relic ? 'success' : 'primary'}
+                >
+                  {relicSlotToMainStatKeys[slotKey].map((mainStatK) => (
+                    <MenuItem
+                      key={mainStatK}
+                      selected={relic?.mainStatKey === mainStatK}
+                      disabled={relic?.mainStatKey === mainStatK}
+                      onClick={() => update({ mainStatKey: mainStatK })}
+                    >
+                      {mainStatK}
+                    </MenuItem>
+                  ))}
+                </DropdownButton>
+                <CardThemed bgt="light" sx={{ p: 1, flexGrow: 1 }}>
+                  <Typography color="text.secondary">
+                    {relic ? relic.mainStatKey : 'Main Stat'}
+                  </Typography>
+                </CardThemed>
+                <Button
+                  onClick={() => update({ lock: !relic?.lock })}
+                  color={relic?.lock ? 'success' : 'primary'}
+                  disabled={!relic}
+                >
+                  {relic?.lock ? <LockIcon /> : <LockOpenIcon />}
+                </Button>
+              </Box>
+            </Grid>
+
```

Comment:
Can we also get a location selector? I have a PR for LocationAutocomplete, you can use this component once it is merged

https://github.com/frzyc/genshin-optimizer/pull/1410

Reply:
PR is merged fyi

Reply:
[8555d5f](https://github.com/frzyc/genshin-optimizer/pull/1401/commits/8555d5f2ea27e6689ac349c172a1a4ef7bdc7178)